yellowbrick/features/pcoords.py

```diff
@@ -225,40 +246,96 @@ def __init__(self, ax=None, features=None, classes=None, normalize=None,
             )
         self.sample = sample
 
+        # Set sample parameters
+        if isinstance(shuffle, bool):
+            self.shuffle = shuffle
+        else:
+            raise YellowbrickTypeError(
+                "`shuffle` parameter must be boolean"
+            )
+        if self.shuffle:
+            if (random_state is None) or isinstance(random_state, int):
+                self.rng_ = RandomState(random_state)
+            elif isinstance(random_state, RandomState):
+                self.rng_ = random_state
+            else:
+                raise YellowbrickTypeError(
+                    "`random_state` parameter must be None, int, or np.random.RandomState"
+                )
+        else:
+            self.rng_ = None
+
         # Visual Parameters
         self.show_vlines = vlines
         self.vlines_kwds = vlines_kwds or {
             'linewidth': 1, 'color': 'black'
         }
 
-    def draw(self, X, y, **kwargs):
+    def fit(self, X, y=None, **kwargs):
         """
-        Called from the fit method, this method creates the parallel
-        coordinates canvas and draws each instance and vertical lines on it.
+        The fit method is the primary drawing input for the
+        visualization since it has both the X and y data required for the
+        viz and the transform method does not.
+
+        Parameters
+        ----------
+        X : ndarray or DataFrame of shape n x m
+            A matrix of n instances with m features
+
+        y : ndarray or Series of length n
+            An array or series of target or class values
+
+        kwargs : dict
+            Pass generic arguments to the drawing method
+
+        Returns
+        -------
+        self : instance
+            Returns the instance of the transformer/visualizer
         """
+
         # Convert from dataframe
         if is_dataframe(X):
             X = X.as_matrix()
+        if is_series(y):
+            y = y.as_matrix()
 
-        # Choose a subset of samples
-        # TODO: allow selection of a random subset of samples instead of head
-
-        if isinstance(self.sample, int):
-            self.n_samples = min([self.sample, len(X)])
-        elif isinstance(self.sample, float):
-            self.n_samples = int(len(X) * self.sample)
-        X = X[:self.n_samples, :]
+        # Subsample
+        X, y = self._subsample(X, y)
 
         # Normalize
         if self.normalize is not None:
             X = self.normalizers[self.normalize].fit_transform(X)
 
-        # Get the shape of the data
+        super(ParallelCoordinates, self).fit(X, y, **kwargs)
+
```

Comment:
Perhaps I'm missing this in the code review pane, but `fit()` should call `draw()` -- am I missing where this is happening?

Reply:
Hehe, I think I got it, but I'm leaving my original comment above in - the super method is calling `draw()` right? Perhaps it would be more clear if we did something like:

```python
        # the super method calls draw and returns self 
        return super(ParallelCoordinates, self).fit(X, y, **kwargs)
```

Reply:
Comment added.  I should have had this in there.